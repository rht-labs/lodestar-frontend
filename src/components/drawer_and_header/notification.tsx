import React, { useState } from 'react';
import { NotificationBadge } from '@patternfly/react-core';
import { BellIcon } from '@patternfly/react-icons';
import { useNotification } from '../../context/notification_context/notification_hook';

export interface NotificationProps {
  onNotificationClick: () => void;
}

export function Notification(props: NotificationProps) {
  const { hasNotification } = useNotification();
  const [isRead, setRead] = useState(!hasNotification);

  const onClick = () => {
    setRead(true);
    props.onNotificationClick();
  };

  return (
    <>
      <NotificationBadge
        isRead={isRead}
        onClick={onClick}
        title="Notifications"
        aria-label="Notifications"
        data-cy={'notification_badge'}
      >
        <BellIcon />
      </NotificationBadge>
    </>
  );
}
