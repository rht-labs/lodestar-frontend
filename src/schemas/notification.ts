import faker from 'faker/locale/en_US';

export enum NotificationType {
  SUCCESS = "success",
  INFO = "info",
  WARNING = "warning",
  DANGER = "danger",
  DEFAULT = "info"
}

export interface Notification {
  title: string,
  message: string,
  type: NotificationType
}

export class Notification {
  title = '';
  message = '';
  type: NotificationType = NotificationType.DEFAULT;

  constructor({title, message, type}: Notification) {
    this.title = title;
    this.message = message;
    this.type = type;
  }

  static allFromFake(): Notification[] {
    return (
      [
        new Notification(
          {
            title: "Default Notification format",
            message: faker.lorem.paragraph(),
            type: NotificationType.DEFAULT
          }),
        new Notification(
          {
            title: "Test Warning",
            message: faker.lorem.paragraph(),
            type: NotificationType.WARNING
          }),
        new Notification(
          {
            title: "Test Success",
            message: faker.lorem.paragraph(),
            type: NotificationType.SUCCESS
          }),
        new Notification(
          {
            title: "Test Danger",
            message: faker.lorem.paragraph(),
            type: NotificationType.DANGER
          }),
        new Notification(
          {
            title: "Test Info",
            message: faker.lorem.paragraph(),
            type: NotificationType.INFO
          }),
      ]
    )
  }
}