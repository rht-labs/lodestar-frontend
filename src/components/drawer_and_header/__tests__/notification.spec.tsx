import React from 'react';
import { Notification } from '../notification';
import { render } from '@testing-library/react';

describe('Notification UI ', () => {
  test('should match snapshot', () => {
    expect(render(<Notification />)).toMatchSnapshot();
  });
});
