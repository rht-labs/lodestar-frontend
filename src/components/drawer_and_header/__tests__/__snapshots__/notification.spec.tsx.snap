// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification UI  should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-disabled="false"
        aria-label="Notifications"
        class="pf-c-button pf-m-plain"
        data-cy="notification_badge"
        data-ouia-component-id="0"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe="true"
        title="Notifications"
        type="button"
      >
        <span
          class="pf-c-notification-badge pf-m-read"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
              transform=""
            />
          </svg>
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-disabled="false"
      aria-label="Notifications"
      class="pf-c-button pf-m-plain"
      data-cy="notification_badge"
      data-ouia-component-id="0"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      title="Notifications"
      type="button"
    >
      <span
        class="pf-c-notification-badge pf-m-read"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 448 512"
          width="1em"
        >
          <path
            d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
            transform=""
          />
        </svg>
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
